code/qcommon/q_platform.h

```diff
@@ -181,7 +183,13 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
 #define PATH_SEP '/'
 
 #if !defined(ARCH_STRING)
-# error ARCH_STRING should be defined by the Makefile
+#if INTPTR_MAX == INT64_MAX
+#define ARCH_STRING "x86_64"
+#elif INTPTR_MAX == INT32_MAX
+#define ARCH_STRING "x86"
+#else
```

Comment:
Hardcode x86 and x86_64 seems like a bad idea as it will just cover up issues.

Reply:
I've reverted everything in q_platform.h and set the ARCH_STRING via cmake script

Reply:
Thanks.